From: Robert Schwebel <r.schwebel@pengutronix.de>
Date: Thu, 14 Jun 2012 11:44:29 +0200
Subject: [PATCH] libffi: fix duplicate case on powerpc 32 bit

Without this fix, gcc complains about a duplicate case value:

../src/powerpc/ffi.c: In function 'ffi_prep_cif_machdep':
../src/powerpc/ffi.c:670:5: error: duplicate case value
../src/powerpc/ffi.c:667:5: error: previously used here

rsc: submitted as http://sourceware.org/ml/libffi-discuss/2012/msg00201.html

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>
---
 src/powerpc/ffi.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/powerpc/ffi.c b/src/powerpc/ffi.c
index baca694..257f02f 100644
--- a/src/powerpc/ffi.c
+++ b/src/powerpc/ffi.c
@@ -664,9 +664,11 @@ ffi_prep_cif_machdep (ffi_cif *cif)
   switch (type)
     {
 #ifndef __NO_FPRS__
+#if FFI_TYPE_LONGDOUBLE != FFI_TYPE_DOUBLE
     case FFI_TYPE_LONGDOUBLE:
       flags |= FLAG_RETURNS_128BITS;
       /* Fall through.  */
+#endif
     case FFI_TYPE_DOUBLE:
       flags |= FLAG_RETURNS_64BITS;
       /* Fall through.  */
